### ISSUE TITLE
Patching data without the old value

### ISSUE THREAD

**Issue by sjoerd222888** [benjamine/jsondiffpatch#92]
I want to create a delta patch manually and for this I do not want to provide the old value.

In a little experiment I see it works if I just provide `null` as old value.

    var testObj = {
         testValue : "Hello World"
    }
    delta = {
        "testValue" : [ null, "Fish" ]
    }
    console.log(testObj.testValue);
    jsondiffpatch.patch(testObj, delta);
    console.log(testObj.testValue);

Is this generally possible with your delta format? Or will I have some issues with this approach?

**Comment by benjamine** [comment]
this is ok, when the old value doesn't match the current value being patched, that would indicate a change conflict, ie. the value you're trying to change, was recently changed by other party, and that's something you want to handle in certain scenarios.

This is equivalent to a 3-way merge, so you have a few ways to solve it, the current one is "last wins", ie. old value is ignored and patch succeeds.
But the old value is there because I plan to add conflict resolution in a future version, when that happen there won't be a breaking change, I'd add a setting to specify the conflict resolution strategy, where the default would be the current one (last wins).

The idea is to have at least these additional options:
- first wins: ignore the new value, not very common but useful in certain scenarios where
- error: on conflict, just throw an error, stopping the patch operation. this is most common in scenarios where data integrity is essential.
- custom: providing a conflict resolution handler (a 3-way merge function), where you could choose any of the 3 values, or do a custom merge.

In short, what you're doing is safe :)
